### PR TITLE
Show roles of a user

### LINKED ISSUE #2657: Show roles of a user
This issue has been migrated from Redmine: https://dev.icinga.com/issues/13637

**Created by ClemensBW on 2016-12-16 10:38:28 +00:00**

Hello,

if I want see user details as admin, for local users are only the icingaweb2 usergroups are visable.

https://$url$/icingaweb2/user/list?page=2#!/icingaweb2/user/show?backend=icingaweb2&user=icingaadmin

Can you please add the icingaweb2 roles to this view too?
(and the option to add this user in a role too, linke the usergroup option :o))

### PR DESCRIPTION
**Pull request by pdorschner**
fixes #2657

### CODE REVIEW

**Comment by lippserd** [comment]
ref/NC/626825

File: application/views/scripts/user/show.phtml [hunk 1]
```diff
@@ -40,6 +40,32 @@ use Icinga\Data\Selectable;
             <th><?= $this->translate('Last modified'); ?></th>
             <td><?= $user->last_modified === null ? '-' : $this->formatDateTime($user->last_modified); ?></td>
         </tr>
+        <tr>
+            <th><?= $this->translate('Role Memberships'); ?></th>
+            <td>
+            <?php $roleObj =  $userObj->getRoles(); ?>
```
Comment (nilmerg) [inline application/views/scripts/user/show.phtml:46]: There's a superfluous space after the equal sign, please remove that.

Comment (nilmerg) [reply application/views/scripts/user/show.phtml:46]: The method `getRoles()` returns an array, so please name the variable `$roles`.

File: application/views/scripts/user/show.phtml [hunk 2]
```diff
@@ -40,6 +40,32 @@ use Icinga\Data\Selectable;
...
+                            $role->getName(),
+                            'role/edit',
+                            ['role' => $role->getName()],
+                            ['title' => sprintf($this->translate('Edit role %s'), $role->getName())]
+                        ) ?>
+                    <?php else: ?>
+                        <?= $role->getName() ?>
+                    <?php endif ?>
+                    </li>
+                <?php endforeach ?>
```
Comment (nilmerg) [inline application/views/scripts/user/show.phtml:62]: Roles allow spaces in their names. Currently there is no separator between multiple roles, without this it's difficult to differentiate the different roles. (Especially if there's one with spaces in its name) Please add a comma as separator between role names.

**Comment by pdorschner** [comment]
Hi,
I updated the PR based on the feedback.

**Pull request merged**